### ISSUE TITLE
[Feature] Unused Signals -> Detection

### ISSUE THREAD

**Issue by jakobjung10** [VHDL-LS/rust_hdl#168]
I would appreciate a  feature, which **detects unused signals, variables...**
In the LS-Client, the user could get feedback by a yellow squiggly line under the signal e.g.

RustHDL already supports **detection of multiply-defined signals**.
Maybe this could be implemented in a similar way.

In how far is this possible?

**Comment by sibeov** [comment]
Bump this feature request!

**Comment by kraigher** [comment]
I have started workin on detecting unused *local* declarations. This will find unused signals. It seems fairly easy to implement without requiring much extra code.

**Comment by kraigher** [comment]
I just released 0.69.0 that includes warning for unused local declarations. This means things that are in package headers will never be unused, since you cannot assume that a package is not used by a downstream project. VHDL would need a public/private concept like rust (or added by a tool via comment pragma) for global unused code analysis to be interesting.

I also added also new configuration setting `is_third_party` for libraries that can be set to `true` to disable unused code detection within that library. The use case is for example if you are using VUnit or OSVVM in your project you do not want to see their internal unused code warnings. This `is_third_party` is currently set on `ieee` and `std` library. Its funny but the `ieee` library contains some dead code, I guess they used notepad.
